src/components/encoding-pane/range-filter-shelf.tsx

```diff
@@ -158,6 +183,10 @@ export class RangeFilterShelfBase extends React.Component<RangeFilterShelfProps,
       dateTimePickerOpen = this.state.maxDateTimePickerOpen;
       dataTimePickerOpenAction = this.toggleMaxDateTimePicker;
     }
+    let timeFormat = true;
+    if (this.props.filter.timeUnit === TimeUnit.YEARMONTHDATE) {
```

Comment:
What about other timeUnit?  

Let's extract a method for any timeUnit specific rules and we can define step for all supported timeUnit and throw error for ones that we have not supported yet?  (This way when we want to add more, we can easily know we need to add it here.  

(Same for other things that you do specifically for timeUnits that we currently support.)